### PR TITLE
fix(usePermission): avoid undefined exception

### LINKED ISSUE #1415: usePermission throws exception in iOS Safari
**What is the current behavior?**

When testing on BrowserStack with a physical iPhone 11 in Safari, `navigator.permissions.query` is throwing an exception within `usePermission`.

image

**Steps to reproduce it and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have extra dependencies other than `react-use`. Paste the link to your [JSFiddle](https://jsfiddle.net) or [CodeSandbox](https://codesandbox.io) example below:**

Sign up for BrowserStack (free) and open an app that uses `usePermission` in Safari.

**What is the expected behavior?**

`usePermission` should probably no-op if the Permissions API is not supported. Alternatively, for each kind of permission, it could try and support what Safari is able to support (notifications, mostly).

See Google polyfill for an example: https://github.com/chromium/permissions.request

**Workaround**

I haven't found one yet since Rule of Hooks states you can't conditionally execute a hook. Polyfilling the APIs that I need (notifications) may be the only alternative that is compatible.

**A little about versions:**  
- _OS_: iOS
- _Browser (vendor and version)_: Safari v13
- _React_: 16.13
- _`react-use`_: 15.1.1
- _Did this worked in the previous package version?_ No

### PR DESCRIPTION
**Pull request by kamranayub**
# Description

Closes #1415 

Avoid throwing an undefined exception at runtime due to missing Permissions API. It would be preferable to wrap/abstract permissions for specific APIs in browsers like Safari but that could be out-of-scope for this hook. At the least, we just want to avoid bombing if the API is missing.

## Screenshots

From iPhone 11 in BrowserStack, Storybook correctly displays a blank string in Safari.

image

Previous code in Storybook:

image

## Type of change

- [x] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] **Breaking change** _(fix or feature that would cause existing functionality to not work as before)_

# Checklist
- [x] Read the [Contributing Guide](https://github.com/streamich/react-use/blob/master/CONTRIBUTING.md)
- [x] Perform a code self-review
- [x] Comment the code, particularly in hard-to-understand areas
- [x] Add documentation
- [x] Add hook's story at Storybook
- [x] Cover changes with tests
- [x] Ensure the test suite passes (`yarn test`)
- [ ] Provide 100% tests coverage
- [x] Make sure code lints (`yarn lint`). Fix it with `yarn lint:fix` in case of failure.
- [x] Make sure types are fine (`yarn lint:types`).

### CODE REVIEW

File: src/usePermission.ts [hunk 1]
```diff
@@ -30,13 +30,16 @@ const usePermission = (permissionDesc: PermissionDesc): State => {
...
+    // Permissions API is not available in every browser
+    if (navigator.permissions && navigator.permissions.query) {
+      navigator.permissions
+        .query(permissionDesc)
+        .then((status) => {
+          permissionStatus = status;
+          changeState();
+        })
+        .catch(noop);
+    }
```
Comment (EL-shadow) [inline src/usePermission.ts:42]: If the API is not supported, then we will not know about it. How about making a special status `unavailable` here?
```
} else {
    permissionStatus = 'unavailable';
    changeState();
}
```

Comment (EL-shadow) [reply src/usePermission.ts:42]: I tried my example and understand that this is not the best solution. 
Better this way
```
} else {
    setState('unavailable');
}
```

File: src/usePermission.ts [hunk 2]
```diff
@@ -30,13 +30,16 @@ const usePermission = (permissionDesc: PermissionDesc): State => {
...
   useEffect(() => {
-    navigator.permissions
-      .query(permissionDesc)
-      .then(status => {
-        permissionStatus = status;
-        changeState();
-      })
-      .catch(noop);
+    // Permissions API is not available in every browser
+    if (navigator.permissions && navigator.permissions.query) {
```
Comment (EL-shadow) [inline src/usePermission.ts:34]: Can be simplified `navigator.permissions?.query`

Comment (kamranayub) [reply src/usePermission.ts:34]: Looks like this was already updated to

```
if ("permissions" in navigator) {
```

**Comment by kamranayub** [comment]
oh no

**Comment by kamranayub** [comment]
This issue is already fixed in `master` so I'm just going to go ahead and close this 🏃

**Pull request closed without merging**